### PR TITLE
422  - Setting up pixl locally

### PR DESCRIPTION
**Pull request by mxochicale**
# Addressing #422 
Mainly, I am trying to setup pixl on my local host device, so I am adding few changes. 

These are some notes that came to mind when looking pixl repo:
  * README looks great and migth be improving by linking relevant sections to avoid duplication (dev and design sections can be moved to https://github.com/UCLH-Foundry/PIXL/tree/main/cli)
  * Do we want to remove versions of deps in https://github.com/UCLH-Foundry/PIXL/blob/main/cli/pyproject.toml (core does not have a version)?
  * Duplicated requirements for test and dev in https://github.com/UCLH-Foundry/PIXL/blob/main/cli/pyproject.toml and https://github.com/UCLH-Foundry/PIXL/blob/main/pixl_core/pyproject.toml and other `pixl_*` packages
  * Duplication of setups instructions: 
    * https://github.com/UCLH-Foundry/PIXL/tree/main?tab=readme-ov-file#development
    * https://github.com/UCLH-Foundry/PIXL/tree/main/cli#installation
    * https://github.com/UCLH-Foundry/PIXL/blob/main/docs/setup/developer.md
  * pytest requires debugging as requires other deps to be run 
      ```
      E   ModuleNotFoundError: No module named 'pixl_imaging'
      E   ModuleNotFoundError: No module named 'pixl_dcmd'
      E   ModuleNotFoundError: No module named 'nibabel'
      2 warnings, 6 errors in 0.58s
      ```

### CODE REVIEW

**Review by stefpiatek — COMMENTED** [review]
Going to wait until #420 is merged before looking at this one, as there's a fair bit of overlap and I think it will also solve your issues

File: .github/ISSUE_TEMPLATE/not_working.md
Comment (stefpiatek) [inline .github/ISSUE_TEMPLATE/not_working.md:2]: this seems like a duplicate of a bug report, with the intention that someone isn't as sure about it? I'd probably aim to remove it in favour of just having one way to report bugs

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (stefpiatek) [inline .github/ISSUE_TEMPLATE/bug_report.md:8]: can you convert this into the yaml style template that we've got for our stories please?

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (stefpiatek) [inline .github/ISSUE_TEMPLATE/feature_request.md:2]: can you also conver this to a yaml style template so that we're being consistent across the repo for our templates 🙏🏻

File: cli/README.md
Comment (stefpiatek) [inline cli/README.md:23]: I wouldn't specify how they set up their environments, as we've given the choice above and its implied

File: cli/README.md
Comment (stefpiatek) [inline cli/README.md:18]: if we want to have some virtual env documentation then I thin this should go into the `docs/dev` directory

**Review by stefpiatek — COMMENTED** [review]
Some early feedback before #420 is merged

File: .github/ISSUE_TEMPLATE/not_working.md
Comment (mxochicale) [reply .github/ISSUE_TEMPLATE/not_working.md]: sorted out here 4ecfdb1

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (mxochicale) [reply .github/ISSUE_TEMPLATE/bug_report.md]: sorted out here 4ecfdb1

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (mxochicale) [reply .github/ISSUE_TEMPLATE/feature_request.md]: sorted out here 4ecfdb1

File: cli/README.md
Comment (mxochicale) [reply cli/README.md]: resolved here 5f00eee

File: cli/README.md
Comment (mxochicale) [reply cli/README.md]: resolved here [917f8c5](https://github.com/UCLH-Foundry/PIXL/pull/423/commits/917f8c5ab462f2df25161b795c3d6ddbabc377d9)

**Review by p-j-smith — CHANGES_REQUESTED** [review]
having a PR template and bug report template is a good idea. A few minor comments - some of the items in the PR template checklist are already enforced by required CI checks, and there are a few commands in the docs that error

File: cli/tests/conftest.py
Comment (p-j-smith) [inline cli/tests/conftest.py:130]: if there's a bug here it's probably best to open a separate issue / PR to fix it, just to keep it distinct from the doc improvements. Also, it looks like `timezone.UCT` was [added in Python 3.11](https://docs.python.org/3/library/datetime.html#datetime.UTC). As the tests run in Python 3.11, perhaps we should suggest upgrading from 3.10?

Comment (mxochicale) [reply cli/tests/conftest.py]: :heavy_check_mark: e6840d53c7fce5157e9cfd4ceac48548cde16e26

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: addressed here b55368d

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: addressed here b55368d

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: addressed here b55368d

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: addressed here b55368d

File: docs/setup/developer.md
Comment (mxochicale) [reply docs/setup/developer.md]: resolved here c86cc5641ab0b0c1ddc2f8184426de21eddfcde7

File: cli/README.md
Comment (mxochicale) [reply cli/README.md]: resolved here c86cc5641ab0b0c1ddc2f8184426de21eddfcde7

File: docs/setup/developer.md
Comment (mxochicale) [reply docs/setup/developer.md]: resolved here c86cc5641ab0b0c1ddc2f8184426de21eddfcde7

File: docs/setup/developer.md
Comment (mxochicale) [reply docs/setup/developer.md]: resolved here c86cc5641ab0b0c1ddc2f8184426de21eddfcde7

File: .github/pull_request_template.md
Comment (stefpiatek) [inline .github/pull_request_template.md:25]: if this is tested, do you need them to ask at the PR stage to check? Might be worth adding to contribution if its not in there instead?

File: .github/pull_request_template.md
Comment (stefpiatek) [inline .github/pull_request_template.md:26]: I'm not quite sure what this means

File: .github/pull_request_template.md
Comment (stefpiatek) [inline .github/pull_request_template.md:27]: we squash merge so probably doesn't need to have the rebase mention

File: CONTRIBUTING.md
Comment (stefpiatek) [inline CONTRIBUTING.md:37]: Id be tempted to make it more explicit that you replace these. maybe surrounding by `<` or `{` for all of the steps

File: CONTRIBUTING.md
Comment (stefpiatek) [inline CONTRIBUTING.md:65]: for these we use squash merge, so we can just pull in the changes and merge the main branch into it (as the PR will only create a single commit in main when merged). To me at least that's easier to talk people through rather than rebasing if its a long lived feature branch

File: CONTRIBUTING.md
Comment (stefpiatek) [inline CONTRIBUTING.md:67]: I don't think there's a next section, but we could point them to making sure that they've followed the developer setup and have precommit installed?

File: cli/README.md
Comment (stefpiatek) [inline cli/README.md:20]: I'm not sure if this was intentional but should probably be `python -m`

File: docs/setup/developer.md
Comment (stefpiatek) [inline docs/setup/developer.md:7]: out of interest, what's the `VE` here?

File: docs/setup/developer.md
Comment (stefpiatek) [inline docs/setup/developer.md:79]: aw no smiley

**Review by stefpiatek — APPROVED** [review]
Looks good, Happy for you to merge once you've addressed all of the comments to your satisfaction

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: resolved here 43d3337e3bb644d554f94296cc23abe7fd250615

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: resolved here 43d3337e3bb644d554f94296cc23abe7fd250615

File: .github/pull_request_template.md
Comment (mxochicale) [reply .github/pull_request_template.md]: resolved here 43d3337e3bb644d554f94296cc23abe7fd250615

File: CONTRIBUTING.md
Comment (p-j-smith) [inline CONTRIBUTING.md:71]:
```suggestion
8. Remove your merged branch from your repo
```

**Review by p-j-smith — APPROVED** [review]
looking good! Just one reference to https://github.com/UCL/real-time-ai-for-surgery that should be removed before merging

File: CONTRIBUTING.md
Comment (mxochicale) [reply CONTRIBUTING.md]: resolved here 16c1f6dea29535000d581234f68de7af72dcbcad

File: CONTRIBUTING.md
Comment (mxochicale) [reply CONTRIBUTING.md]: resolved here dbfe2042d9df3ec1e6d7270b299ec6035660e0fd

File: CONTRIBUTING.md
Comment (stefpiatek) [reply CONTRIBUTING.md:37]: I guess could imagine someone coming the the project who is inexperienced and assuming that if they were working on issue 422 "fix flabber". Then they'd make a branch `422-branch-name`, but if you don't thinks that's likely I'm not too bothered by it

File: cli/README.md
Comment (mxochicale) [reply cli/README.md]: sorted out here e30ee053808506446244759de74d6fe4dee229eb

File: docs/setup/developer.md
Comment (mxochicale) [reply docs/setup/developer.md:7]: sorted our here e30ee053808506446244759de74d6fe4dee229eb

File: docs/setup/developer.md
Comment (mxochicale) [reply docs/setup/developer.md:79]: Sorry, we can always use some [emojis](https://gist.github.com/rxaviers/7360908) in the sections names and removed that one as it was at the end of the sentence and in the middle of the readme file

File: CONTRIBUTING.md
Comment (mxochicale) [reply CONTRIBUTING.md]: Thanks, just added changes here 01c71b12bc564aaf722f52420ce85b35476feeb8

File: CONTRIBUTING.md
Comment (mxochicale) [reply CONTRIBUTING.md]: sorted out here 4ce2a2ebc48f2e445a9af4896920c9ea15a65221

**Pull request merged**
